### PR TITLE
Add NAME_FIXED option for rdma_rename

### PR DESCRIPTION
**Pull request by edwardsp**
usage:  ```rdma_rename <old-name> NAME_FIXED <new-name>```

### CODE REVIEW

**Review by rleon — CHANGES_REQUESTED** [review]
Please add this mode in the first block of this file where all possible modes are listed.
Also please add description to the commit message too.

It will be also nice to add commented example line to our udev rule.

Thanks, it is very welcomed change.

File: kernel-boot/rdma_rename.c [hunk 1]
```diff
@@ -483,6 +483,19 @@ static int by_guid(struct data *d)
 	return 0;
 }
 
+static int set_fixed_name(struct data *d, char *name)
+{
+	int ret = -1;
```
Comment (rleon) [inline kernel-boot/rdma_rename.c:488]: No need in assignment, you are overwriting it.

File: kernel-boot/rdma_rename.c [hunk 2]
```diff
@@ -629,6 +650,8 @@ int main(int argc, char **argv)
 		ret = by_pci(&d);
 	if (ret && (np & NAME_GUID))
 		ret = by_guid(&d);
+	if (ret && (np & NAME_FIXED))
```
Comment (rleon) [inline kernel-boot/rdma_rename.c:653]: It is not correct if (..), if everything above failed, we will execute this section which is not what we want.

You need to place NAME_FIXED outside of this if .. block.

Comment (edwardsp) [reply kernel-boot/rdma_rename.c:653]: I was just trying to keep with the convention of the current code.  NAME_FIXED can only be set if the other options aren't set (in the current implementation).  By "and"ing with the return value we are just setting an order of precedence if this ever were to change.  Would you prefer it to be, `if (np & NAME_FIXED)`?  (it should be checked for the `if(ret) goto out;` part)

File: kernel-boot/rdma_rename.c (hunk 1 above)
Comment (edwardsp) [reply kernel-boot/rdma_rename.c]: thanks, I've updated it

File: kernel-boot/rdma_rename.c (hunk 2 above)
Comment (rleon) [reply kernel-boot/rdma_rename.c:654]: Please squash all commits into one and push with "-f" to this topic.

The "ret ... " scheme is needed for the fall-back, but you are right, we can leave it here.
Sorry for my mistake.

Comment (edwardsp) [reply kernel-boot/rdma_rename.c:654]: I've squashed the commits.  Thanks

Comment (rleon) [reply kernel-boot/rdma_rename.c:654]: 1. Your commit message is still empty.
2. Update and add an example to kernel-boot/rdma-persistent-naming.rules as a comment, something like this:
#ACTION=="add", SUBSYSTEM=="infiniband", PROGRAM="rdma_rename %k NAME_FIXED static_name"

Comment (edwardsp) [reply kernel-boot/rdma_rename.c:654]: I've updated both.  Thanks!

**Comment by rleon** [comment]
Thanks

**Pull request merged**